frontend/packages/ceph-storage-plugin/src/components/attached-devices-mode/lso-disk-inventory/ocs-disks-list.tsx

```diff
@@ -0,0 +1,277 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import * as cx from 'classnames';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useSelector } from 'react-redux';
+import {
+  Table,
+  TableProps,
+  TableRow,
+  TableData,
+  RowFunction,
+} from '@console/internal/components/factory';
+import { sortable } from '@patternfly/react-table';
+import { humanizeBinaryBytes, Kebab } from '@console/internal/components/utils';
+import { usePrometheusPoll } from '@console/internal/components/graphs/prometheus-poll-hook';
+import { PrometheusEndpoint } from '@console/internal/components/graphs/helpers';
+import {
+  osdDiskInfoMetric,
+  DATA_RESILIENCY_QUERY,
+  StorageDashboardQuery,
+} from '@console/ceph-storage-plugin/src/constants/queries';
+import { TemplateInstanceModel } from '@console/internal/models';
+import { TemplateInstanceKind } from '@console/internal/module/k8s';
+import { PrometheusResult } from '@console/internal/components/graphs';
+import {
+  useK8sWatchResource,
+  WatchK8sResource,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import { RootState } from '@console/internal/redux';
+import { NotificationAlerts } from '@console/internal/reducers/ui';
+import { Alert } from '@console/internal/components/monitoring/types';
+import { DiskMetadata } from '@console/local-storage-operator-plugin/src/components/disks-list/types';
+import {
+  NodesDisksListPage,
+  NodesDisksListPageProps,
+} from '@console/local-storage-operator-plugin/src/components/disks-list/disks-list-page';
+import {
+  CEPH_STORAGE_NAMESPACE,
+  OSD_DOWN_ALERT,
+  OSD_DOWN_AND_OUT_ALERT,
+} from '../../../constants/index';
+import { OCSKebabOptions } from './ocs-kebab-options';
+import { OCSStatus } from './ocs-status-column';
+import {
+  ActionType,
+  OCSColumnState,
+  initialState,
+  reducer,
+  OCSDiskList,
+  OCSColumnStateAction,
+  Status,
+  OCSDiskStatus,
+} from './state-reducer';
+
+const getTiBasedStatus = (status: string): OCSDiskStatus => {
+  switch (status) {
+    case 'Not Ready':
+      return Status.PreparingToReplace;
+    case 'Ready':
+      return Status.ReplacementReady;
+    case 'Failed':
+      return Status.ReplacementFailed;
+    default:
+      return null;
+  }
+};
+
+const getAlertsBasedStatus = (alertName: string): OCSDiskStatus => {
+  switch (alertName) {
+    case OSD_DOWN_ALERT:
+      return Status.NotResponding;
+    case OSD_DOWN_AND_OUT_ALERT:
+      return Status.Offline;
+    default:
+      return null;
+  }
+};
+
+const getTiResource = (): WatchK8sResource => {
+  return {
+    kind: TemplateInstanceModel.kind,
+    namespace: CEPH_STORAGE_NAMESPACE,
+    isList: true,
+  };
+};
```

Comment:
```suggestion
const getTiResource = (): WatchK8sResource => ({
    kind: TemplateInstanceModel.kind,
    namespace: CEPH_STORAGE_NAMESPACE,
    isList: true,
  });
```
unnecessary `return`